ui/raidboss/data/05-shb/raid/e6s.js

```diff
@@ -7,6 +7,113 @@
   },
   timelineFile: 'e6s.txt',
   triggers: [
+    {
+      id: 'E6S Strike Spark',
+      regex: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexDe: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexFr: Regexes.startsUsing({ source: ['Ifrit', 'Raktapaksa'], id: '4BD3', capture: false }),
+      regexJa: Regexes.startsUsing({ source: ['イフリート', 'ラクタパクシャ'], id: '4BD3', capture: false }),
+      promise: function(data) {
+        let p = new Promise(async (res) => {
+          // helper function to delay the promise execution for the given time
+          const sleep = (m) => new Promise((r) => setTimeout(r, m));
+
+          await sleep(10000);
+
+          let combatantNames = null;
+
+          const ifritLocaleNames = {
+            en: 'Ifrit',
+            de: 'Ifrit',
+            fr: 'Ifrit',
+            ja: 'イフリート',
+          };
+
+          const raktapaksaLocaleNames = {
+            en: 'Raktapaksa',
+            de: 'Raktapaksa',
+            fr: 'Raktapaksa',
+            ja: 'ラクタパクシャ',
+          };
+
+          // select the 4 most recent Ifrit or Raktapaksa's depending on phase
+          if (data.phase === 'ifrit')
+            combatantNames = { names: [ifritLocaleNames[data.lang]] };
+          else
+            combatantNames = { names: [raktapaksaLocaleNames[data.lang]] };
+
+
+          let combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            data: combatantNames,
+          });
+
+          // if we could not retrieve combatant data, the
+          // trigger will not work, so just resume promise here.
+          if (!(combatantData !== null &&
+            combatantData.combatants &&
+            combatantData.combatants.length)) {
+            data.safeZone = null;
+            res();
+          }
+
+          // we need to filter for the Ifrit with the highest HEX ID
+          // since that one is always the safe spot.
+          let currentHighestCombatant =
+            combatantData.combatants.sort((a, b) => a.ID.localeCompare(b.ID)).pop();
+
+          // all variation ranges for all the 9 ball positions for the kicking actors
+          // north      x: 96-104   y: 85-93
+          // northeast  x: 107-115  y: 85-93
+          // northwest  x: 85-93    y: 85-93
+          // east       x: 107-115  y: 96-104
+          // west       x: 85-93    y: 96-104
+          // south      x: 96-104   y: 107-115
+          // southeast  x: 107-115  y: 107-115
+          // southwest  x: 85-93    y: 107-115
+          let safeZoneObj1 = { en: '', de: '' };
+          let safeZoneObj2 = { en: '', de: '' };
+
+          // don't need to go through all the posibilities,
+          // only those 4 ifs do reflect the above positions
+          if (currentHighestCombatant.posY > 84 && currentHighestCombatant.posY < 94) {
+            safeZoneObj1 = {
+              en: 'north',
```

Comment:
It might make sense to have common phrases like north/south/east/west/etc be in a new helper class in the same vein as the `Regexes` class? At the least, it makes more sense to have this be a const struct like this instead, which can have the correct return value looked up from, so that all the localization is in one place:
```javascript
const directions = {
  nw: {
    en: 'northwest',
    de: 'nordwesten',
    ...
  },
  n: {
    ...
  }
  ...
};
```

Reply:
I need to think about where this would go.  I think `responses.js` itself is meant to be more of a "this is the entire trigger's response, with maybe some logic", but maybe it'd be good to have some file of localized phrases on their own. What would I call such a file... :thinking:

Reply:
Hmm... `language.js`? Could export a module such that:
```javascript
let Language = {
  directions: {
    nw: ...
  },
};
if (typeof module !== 'undefined' && module.exports) {
  module.exports = {
    language: Language,
  };
}
```

I'm just not sure what else would go in the file, given that a lot of the localization is already present in `responses.js` as it is.

Reply:
This is the last bit of the PR I didn't adjust yet. Any more input or should I ignore it for now?

Reply:
I'm still of the opinion that it makes more sense to have all the language stuff in one spot in the trigger and then just map to it for the direction, and it would make it easier to relocate later to a common module.

Reply:
Maybe we can just go with `responses.js` and add entries such as `Responses.goNorth()` etc? Would create a bit more writing work in the trigger, but yeah, having the language stuff in one spot sounds better to me aswell.

Reply:
This PR seems relevant to the conversation.
https://github.com/quisquous/cactbot/pull/1303

Reply:
Belatedly, I think this is fine to ignore and fix in a followup.

I agree with valarnin that translation-wise, it's almost always better to not do any string concatenation.  To just add logic that figures out that something is northeast, and then lookup "northeast" in some map.

Re: where that map should live, I think the Responses object itself is not the best fit, because Responses.foo is a function that returns an object meant to be attached to the `response` field of a trigger, i.e. `{ alertText: { en: 'Buster on YOU' } }` where this is more something where you just want `Trans.north` to literally be the value `{ en: 'North', etc }`.  If that makes sense?  Responses is good for simple and complicated trigger responses, but I think there's room for something simpler.

Reply:
Filed #1307